Source/WebCore/platform/graphics/cg/GraphicsContextCG.cpp

```diff
@@ -832,9 +833,15 @@ void GraphicsContextCG::fillRect(const FloatRect& rect, Gradient& gradient, cons
         gradient.paint(layerContext);
         CGContextDrawLayerInRect(context, rect, layer.get());
     } else {
-        CGContextClipToRect(context, rect);
+        bool skipClip = getCTM().mapRect(rect).contains(CGContextGetDeviceClipBoundingBox(context));
```

Comment:
Why is this not `CGContextGetClipBoundingBox()` ?

Reply:
And you need to explain, either in a comment or the commit message, why mapping the rect through the current CTM is the right thing to do. Could you compare the unmapped rect with `CGContextGetClipBoundingBox()`? If not, why not?

Reply:
Indeed those two should be equivalent, I will try that instead